lib/iris/_lazy_data.py

```diff
@@ -260,26 +262,37 @@ def as_lazy_data(data, chunks=None, asarray=False, dims_fixed=None):
         but reduced by a factor if that exceeds the dask default chunksize.
 
     """
-    if chunks is None:
-        # No existing chunks : Make a chunk the shape of the entire input array
-        # (but we will subdivide it if too big).
-        chunks = list(data.shape)
-
-    # Adjust chunk size for better dask performance,
-    # NOTE: but only if no shape dimension is zero, so that we can handle the
-    # PPDataProxy of "raw" landsea-masked fields, which have a shape of (0, 0).
-    if all(elem > 0 for elem in data.shape):
-        # Expand or reduce the basic chunk shape to an optimum size.
-        chunks = _optimum_chunksize(
-            chunks, shape=data.shape, dtype=data.dtype, dims_fixed=dims_fixed
-        )
-
+    if as_dask:
+        if chunks is not None:
+            raise ValueError(
+                f"Dask chunking chosen, but chunks already assigned value {chunks}"
+            )
+    else:
+        if chunks is None:
+            # No existing chunks : Make a chunk the shape of the entire input array
+            # (but we will subdivide it if too big).
+            chunks = list(data.shape)
+
+        # Adjust chunk size for better dask performance,
+        # NOTE: but only if no shape dimension is zero, so that we can handle the
+        # PPDataProxy of "raw" landsea-masked fields, which have a shape of (0, 0).
+        if all(elem > 0 for elem in data.shape):
+            # Expand or reduce the basic chunk shape to an optimum size.
+            chunks = _optimum_chunksize(
+                chunks,
+                shape=data.shape,
+                dtype=data.dtype,
+                dims_fixed=dims_fixed,
+            )
     if isinstance(data, ma.core.MaskedConstant):
         data = ma.masked_array(data.data, mask=data.mask)
     if not is_lazy_data(data):
-        data = da.from_array(
-            data, chunks=chunks, asarray=asarray, meta=np.ndarray
-        )
+        if as_dask:
+            data = da.from_array(data, asarray=asarray, meta=np.ndarray)
+        else:
+            data = da.from_array(
+                data, chunks=chunks, asarray=asarray, meta=np.ndarray
+            )
```

Comment:
>In _lazy_data, there are two checks for as_dask, at `263` and `283`. 
>This *might* not be needed, if the check at `280` could be done before `275`. 

If you're interested in efficiency then the neatest approach IMO would be to contain `asarray` and `meta` in a `dict` - constructed at the top of the function - then optionally add `chunks` to this `dict` in the correct place.

Then you can just have a single call to `da.from_array()`:

```python
da.from_array(data, **my_dict)
```